spec/immutable-data-structures.html

```diff
@@ -96,7 +61,9 @@ <h1>Tuple ( ..._items_ )</h1>
           1. If NewTarget is not *undefined*, throw a *TypeError* exception.
           1. Let _elements_ be a List containing the arguments passed to this function.
           1. Let _length_ to be the number of elements in _elements_.
-          1. Let _tuple_ be ? CreateTupleFromList(_elements_)
+          1. For each element _e_ of _elements_,
+            1. If Type(_e_) is Object, throw a *TypeError* exception.
+          1. Let _tuple_ be a new Tuple value whose [[Sequence]] is _elements_.
           1. Assert: The value of _tuple_'s *"length"* property is _length_.

```

Comment:
BTW this is pedantically "wrong" since primitives don't have properties. I guess you meant, Assert: ! GetV(_tuple_, *"length"*) is _length_.

Reply:
Good point. The question really is, do we even need this check? I guess it seems superfluous, constructing a Tuple from a list will always result in a Tuple whose length is the length of said list.

Reply:
I'm fine with removing the assertion too.